### PR TITLE
Counter

### LINKED ISSUE #38: feat: First contract (counter)
Implement this contract in Cairo
https://solidity-by-example.org/first-app/

### PR DESCRIPTION
**Pull request by HansBhatia**
Closes #38

### CODE REVIEW

**Comment by HansBhatia** [comment]
@enitrat Ready for review!

File: listings/ch00-introduction/counter/src/counter.cairo [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        let counter: u256 = _counter::read() + 1;
+        _counter::write(counter);
+    }
+
+    #[external]
+    fn decrement() {
+        // Store counter value - 1
+        let counter: u256 = _counter::read() - 1;
+        _counter::write(counter);
+    }
```
Comment (enitrat) [inline listings/ch00-introduction/counter/src/counter.cairo:23]: can you add a view function `get_counter` that returns the counter's value?

File: src/ch00-11-counter.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# Simple Counter
+
+This is the Cairo adaptation of the [Solidity by example First Application](https://solidity-by-example.org/first-app/).
```
Comment (enitrat) [inline src/ch00-11-counter.md:3]: I think we can skip the credits here, there's nothing very specific about a counter contract that would require it

**Review by enitrat — APPROVED** [review]
Great work 🙌

**Comment by HansBhatia** [comment]
@enitrat Changes made 👍

**Review by enitrat — APPROVED** [review]

**Pull request merged**